Tests/Shared.Specs/FakeClock.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+
+namespace FluentAssertions.Specs
+{
+    /// <summary>
+    /// Implementation of <see cref="IClock"/> for testing purposes only.
+    /// </summary>
+    /// <remarks>
+    /// It allows you to control the "current" time.
+    /// </remarks>
+    internal class FakeClock : IClock
+    {
+        private DateTime fakeNow = DateTime.Now;
```

Comment:
🔧 How about working entirely with `TimeSpan`, starting with `TimeSpan.Zero`?
The difference between two `TimeSpan` is also a `TimeSpan` as required for `ITimer.Elapsed`.